ethereum/eth/src/main/java/tech/pegasys/pantheon/ethereum/eth/messages/TransactionsMessageLimiter.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.eth.messages;
+
+import tech.pegasys.pantheon.ethereum.rlp.BytesValueRLPOutput;
+
+public interface TransactionsMessageLimiter {
```

Comment:
(optional) One other idea to simplify the class structure would be to make this a static internal class of `LimitedTransactionsMessages` marked as a `@FunctionalInterface`, and keep the default limiter (`DefaultTransactionMessageLimiter`) as a static field on `LimitedTransactionsMessages`.  You could then have a method `createLimited(transactions)` that uses the default limiter.  This means fewer files and fewer dependencies that have to be passed around.

Reply:
I have opted for this idea. Indeed it simplifies the design.